partial/pandas/_libs/interval.pyi

```diff
@@ -29,43 +40,59 @@ class IntervalMixin:
     @property
     def is_empty(self: IntervalMixinProtocol) -> bool: ...
 
-class Interval(IntervalMixin, Generic[Orderable]):
+class Interval(IntervalMixin, Generic[OrderableT]):
     @overload
-    def left(self: Interval[OrderableScalar]) -> OrderableScalar: ...
+    def left(self: Interval[OrderableScalarT]) -> OrderableScalar: ...
     @overload
-    def left(self: Interval[OrderableTimes]) -> OrderableTimes: ...
+    def left(self: Interval[OrderableTimesT]) -> OrderableTimes: ...
     @property
     @overload
-    def left(self: Interval[Orderable]) -> Orderable: ...
+    def left(self: Interval[OrderableT]) -> Orderable: ...
```

Comment:
Are all the overloads needed for `left`/`right`? I think this might be enough:
```py
def left(self: Interval[OrderableT]) -> OrderableT: ...
def right(self: Interval[OrderableT]) -> OrderableT: ...
```

Reply:
> Are all the overloads needed for `left`/`right`? I think this might be enough:
> 
> ```python
> def left(self: Interval[OrderableT]) -> OrderableT: ...
> def right(self: Interval[OrderableT]) -> OrderableT: ...
> ```

Yes, that is correct.  Changed in next commit.